### PR TITLE
refactor(core): move TransferState implementation to core

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit moves the `TransferState` class implementation to `@angular/core`. This class will be needed in `core` in followup changes (💧) and `core` can not depend on `platform-browser` package.

The `core` package does **not** export those symbols into public API surface. The `platform-browser` package re-exports the symbols under same names for backwards-compatibility.

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: goldens/public-api/platform-browser/index.md [hunk 1]
```diff
@@ -227,21 +225,7 @@ export class Title {
...
-    onSerialize<T>(key: StateKey<T>, callback: () => T): void;
-    remove<T>(key: StateKey<T>): void;
-    set<T>(key: StateKey<T>, value: T): void;
-    toJson(): string;
-    // (undocumented)
-    static ɵfac: i0.ɵɵFactoryDeclaration<TransferState, never>;
-    // (undocumented)
-    static ɵprov: i0.ɵɵInjectableDeclaration<TransferState>;
-}
+export { TransferState }
```
Comment (AndrewKushnir) [inline goldens/public-api/platform-browser/index.md:228]: That's unfortunate that we've lost typing info here, but we should consider exposing `TransferState` from the `@angular/core` instead in followup PRs (also re-exporting from the `platform-browser` for backwards-compatibility).

Comment (alan-agius4) [reply goldens/public-api/platform-browser/index.md:228]: I was going to ask about this, should we make the exports from `@angular/core` as non private and deprecate the exports in `@angular/platform-browser`.

**Review by alan-agius4 — APPROVED** [review]
LGTM. Did have a question about the exports though.

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: fw-core, fw-platform-server, integration-tests, public-api

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:512689227:BASE:512689268:1677527372048:dd7d674).

File: goldens/public-api/platform-browser/index.md (hunk 1 above)
Comment (AndrewKushnir) [reply goldens/public-api/platform-browser/index.md:228]: Discussed offline: we'll land this change as is first (so there are no changes to the public API) and update the public API after that if needed.

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 459fbb31d1f70c8c639f795759cee4daaaf44cd0.

**Pull request closed without merging**